featuregates/controller/pkg/feature/feature_suite_test.go

```diff
@@ -0,0 +1,565 @@
+		}, timeout, interval).Should(BeTrue())
+
+		Expect(k8sClient.Delete(ctx, feature)).Should(BeNil())
+		Expect(k8sClient.Delete(ctx, featureGate)).Should(BeNil())
+	})
+
+	It("Should activate stable features by default and stable features should be immutable", func() {
+		feature := getTestFeature()
+		feature.Spec.Stability = corev1alpha2.Stable
+		Expect(k8sClient.Create(ctx, feature)).Should(Succeed())
+
+		Eventually(func() bool {
+			err := k8sClient.Get(ctx, types.NamespacedName{Name: feature.Name}, feature)
+			return err == nil
+		}, timeout, interval).Should(BeTrue())
+
+		Eventually(func() bool {
+			err := k8sClient.Get(ctx, types.NamespacedName{Name: feature.Name}, feature)
+			return err == nil && feature.Status.Activated == true
+		}, timeout, interval).Should(BeTrue())
+
+		featureGate := getTestFeatureGate()
+		featureGate.Spec.Features = append(featureGate.Spec.Features, corev1alpha2.FeatureReference{
+			Name:     feature.Name,
+			Activate: false,
+		})
+		Expect(k8sClient.Create(ctx, featureGate)).ShouldNot(Succeed())
+
+		Expect(k8sClient.Delete(ctx, feature)).Should(BeNil())
+	})
+
+	It("Should activate deprecated features by default and stable features should be mutable", func() {
+		feature := getTestFeature()
+		feature.Spec.Stability = corev1alpha2.Deprecated
+		Expect(k8sClient.Create(ctx, feature)).Should(Succeed())
+
+		Eventually(func() bool {
+			err := k8sClient.Get(ctx, types.NamespacedName{Name: feature.Name}, feature)
+			return err == nil
+		}, timeout, interval).Should(BeTrue())
+
+		Eventually(func() bool {
+			err := k8sClient.Get(ctx, types.NamespacedName{Name: feature.Name}, feature)
+			return err == nil && feature.Status.Activated == true
+		}, timeout, interval).Should(BeTrue())
+
+		featureGate := getTestFeatureGate()
+		featureGate.Spec.Features = append(featureGate.Spec.Features, corev1alpha2.FeatureReference{
+			Name:     feature.Name,
+			Activate: false,
+		})
+		Expect(k8sClient.Create(ctx, featureGate)).Should(Succeed())
+
+		Eventually(func() bool {
+			err := k8sClient.Get(ctx, types.NamespacedName{Name: featureGate.Name}, featureGate)
+			return err == nil
+		}, timeout, interval).Should(BeTrue())
+
+		Eventually(func() bool {
+			err := k8sClient.Get(ctx, types.NamespacedName{Name: featureGate.Name}, featureGate)
+			return err == nil && len(featureGate.Status.FeatureReferenceResults) == 1
+		}, timeout, interval).Should(BeTrue())
+
+		Expect(featureGate.Status.FeatureReferenceResults[0].Status).Should(Equal(corev1alpha2.FeatureReferenceStatus("Applied")))
+
+		Eventually(func() bool {
+			err := k8sClient.Get(ctx, types.NamespacedName{Name: feature.Name}, feature)
+			return err == nil && feature.Status.Activated == false
+		}, timeout, interval).Should(BeTrue())
+
+		Expect(k8sClient.Delete(ctx, feature)).Should(BeNil())
+		Expect(k8sClient.Delete(ctx, featureGate)).Should(BeNil())
+	})
+
+	It("Should remove feature from featuregate status when reference is removed", func() {
```

Comment:
this test should also check if the status of the feature is the default activation state

Reply:
Done